### ISSUE TITLE
Support PDI 6.x

### ISSUE THREAD

**Issue by matthewtckr** [bcolas/PDISforceBulkPlugin#1]
Pentaho Kettle refactored the Salesforce steps into a separate plugin.  This step should be updated to resolve dependencies correctly.

pentaho/pentaho-kettle@fa4b01a5cf00b644b51c3beb288b9b81aa25456f

**Comment by bcolas** [comment]
Will look at this to publish a 6.x compatible version

**Comment by karilint** [comment]
Hi, do you have any schedule estimate for PDI 6.1? In addition, it would be great to be able to query all records (incl. deleted ones).
Keep on good work!

**Comment by ebspeter** [comment]
This is a great feature, it would mean a lot if you could update to 6.x compatible. I would like to contribute but not sure i have the ability.

**Comment by sbadola** [comment]
Hi

I am trying to use latest Salesforce Bulk Plugin in PDI 6.1.0.2 version but it keeps giving me below  error on UI as soon as I drop the Salesforce Bulk Input icon to my transormation

Error -

![image](https://cloud.githubusercontent.com/assets/25355333/22311560/fe68138e-e321-11e6-8782-b04763a9bee0.png)

Can you pls help with this issue.

Thanks
Sanjay

**Comment by codek** [comment]
Same error on PDI 7.x..  Is it just a case of rebuilding with a different pdi dependency version, or something more sinister?

**Comment by matthewtckr** [comment]
I've made some progress in building a 6.1 compatible release, but I'm seeing CSVParseException errors when retrieving some data from Salesforce.

The solution for this that I was working towards was simply to embed the salesforce plugin as a dependency in this plugin.  As noted in the original comment, the Salesforce classes in PDI were moved from the Kettle Engine into a plugin, which caused this issue.

I can't commit to finishing this at the moment, but wanted to share in case someone is able to take it the rest of the way.

https://github.com/matthewtckr/PDISforceBulkPlugin/tree/pdi61compat

Note that PDI 7.1 (7.0 maybe?) changed from using the salesforce-partner jar to the Force-WSC jar, so there may be additional refactoring needed to make this compatible for 7.x.  Hopefully 6.1 is a good small step to take first, though.

**Comment by bcolas** [comment]
Thanks Matt for your help,

I really have difficulties understanding build and dependencies management with ivy, maven, ...
I committed my current build.xml, ... files but the result is not what I am expecting.

@ebspeter @sbadola To solve the problem, I just copy the plugins\pdi-salesforce-plugin\pdi-salesforce-plugin-6.1.jar jar file in plugins\kettle-sforcebulk-input\lib directory.

The test I realized on my company Saleforce environment did not show any CSVParseException errors.
